src/include/detail/linalg/tdb_matrix_with_ids.h

```diff
@@ -188,11 +188,12 @@ class tdbBlockedMatrixWithIds
 
     std::string attr_name = attr.name();
     tiledb_datatype_t attr_type = attr.type();
-    if (attr_type != tiledb::impl::type_to_tiledb<T>::tiledb_type) {
+    if (attr_type != tiledb::impl::type_to_tiledb<IdsType>::tiledb_type) {
```

Comment:
I generally have a practice of using an indirection for types used within a class / function template, rather than the function template name.  Especially for classes, this lets you have uniform identifier names, even when you don't have templated types.  Hence, `using feature_type = T; using ids_type = IdsType` and so on.  It also localizes what you would have to change if you decided to do something different to determine what IdsType is.  Just a suggestion.

Reply:
Sure, updated.